src/spikeinterface/core/waveforms_extractor_backwards_compatibility.py

```diff
@@ -446,6 +446,10 @@ def _read_old_waveforms_extractor_binary(folder, sorting):
     else:
         rec_attributes["probegroup"] = None
 
+    if "dtype" not in rec_attributes:
+        warnings.warn("dtype not found in rec_attributes. Setting to float32")
+        rec_attributes["dtype"] = "float32"
+
```

Comment:
My only question is how often would things be float32? Is the assumption people would have run preprocessing so the majority would have already been converted at some step? I'm just wondering if there is a way to know? 

Or maybe this isn't really used and so the value doesn't really matter?

I'm not sure. Just curious how you're thinking about this.

Reply:
Thanks @zm711 

I think this is only used when setting a temporary recording. Maybe it would be better to not set it and skip the check if non-existing? Then it can be set from the temporary recording object in memory!

Reply:
@zm711 changed strategy to improve and add options to the `do_recording_attributes_match`. The only real need in terms of back-compatibility is to only test for `dtype` if the `rec_attributes` have it, which was not the case for old waveform extractors loaded as analyzers